src/mysky.ts

```diff
@@ -394,26 +706,23 @@ export class MySky {
         return;
       }
 
-      // Parse the seed.
-      const seed = new Uint8Array(JSON.parse(newValue));
-
-      // Launch the new permissions provider.
-      this.permissionsProvider = launchPermissionsProvider(seed);
-
-      // If the email is found, then set up auto-login on Main MySky.
-      const email = localStorage.getItem(EMAIL_STORAGE_KEY);
-      if (email) {
-        // Clear the stored email.
-        //
-        // The email can be cleared here because `localStorage` is only used to
-        // marshal the email from MySky UI over to the invisible MySky iframe.
-        // We don't clear the seed because we need it in storage so that users
-        // are automatically logged-in, when possible. But for the email, it
-        // should be stored on MySky, as the local storage can get cleared,
-        // users can move across browsers etc.
-        localStorage.removeItem(EMAIL_STORAGE_KEY);
-
-        // TODO: Set up auto-login.
+      // Clear any existing value to make sure the storage event is triggered
+      // when we set the key.
+      localStorage.removeItem(PORTAL_LOGIN_COMPLETE_SENTINEL_KEY);
+
+      try {
+        // Parse the seed.
+        const seed = new Uint8Array(JSON.parse(newValue));
+
+        await this.loginFromUi(seed);
+
+        // Signal to MySky UI that we are done.
+        localStorage.setItem(PORTAL_LOGIN_COMPLETE_SENTINEL_KEY, "1");
```

Comment:
nit: This `1` is the only acceptable value for this key. That's confusing. It would be better to put it in a const, so we know that there is only one valid sentinel value.